### PR TITLE
Update grid to be mobile or desktop first

### PR DESCRIPTION
**Pull request by dawnerd**
Option to be mobile first. Defaults to desktop.

There is one thing Im banging my head against the wall with. Classes like col-md-* have to behave slightly differently depending on which type is chosen. I have it working with desktop first now but Im not sure it'll work the other way around without a lot of code. Would love some ideas.

Also added an example page

### CODE REVIEW

**Comment by jgallen23** [comment]
why do md-* need to be different?

File: conf/default-core.yaml [hunk 1]
```diff
@@ -3,3 +3,4 @@ core:
   minimize: false
   dist: '{{CWD}}/.dist'
   assetPath: '{{CWD}}'
+  mobileFirst: false
```
Comment (jgallen23) [inline conf/default-core.yaml]: whats the difference between mobileFirst and default: true?

File: styles/helpers/grid.css [hunk 2]
```diff
@@ -73,6 +69,7 @@
   min-height: 1px;
   padding-left: 15px;
   padding-right: 15px;
+  float: left;
```
Comment (jgallen23) [inline styles/helpers/grid.css]: this seems risky to float all rows. whats the need for it?

File: conf/default-core.yaml (hunk 1 above)
Comment (dawnerd) [inline conf/default-core.yaml]: default: true determines whats the base breakpoint so we're not hardcoding it. mobileFirst handles changing the min/max-width

**Comment by chrisopedia** [comment]
I'm not sure I understand the issue

File: conf/default-core.yaml (hunk 1 above)
Comment (jgallen23) [inline conf/default-core.yaml]: would you ever change one and not the other?

**Comment by dawnerd** [comment]
Before I made the change all of the col-md, col-sm, col-lg etc were based on min width and were mobile first. Now that it can be reversed, col-md doesn't work the same since it falls outside of the breakpoint.

**Comment by chrisopedia** [comment]
The only thing that should change in this scenario is the min width to max and the solves it. The media query shouldn't be invoked on a browser size outside the min width we call desktop

**Comment by jgallen23** [comment]
md==tablet?

File: conf/default-core.yaml (hunk 1 above)
Comment (dawnerd) [inline conf/default-core.yaml]: How would it know which is the default breakpoint? We'd have to sort all the breakpoints by size and compare vs just checking the flag.

Comment (jgallen23) [inline conf/default-core.yaml]: good enough reasoning for me. thanks for clarifying

Comment (dawnerd) [inline conf/default-core.yaml]: I guess we could also just rename the key to be default like you did on typography

File: conf/default-breakpoints.yaml [hunk 3]
```diff
@@ -1,18 +1,23 @@
 breakpoints:
   desktop-wide:
+    max-width: 100vw
     min-width: 1440px
     content: 1440px
     col: xl
+    default: true
```
Comment (jgallen23) [inline conf/default-breakpoints.yaml]: what about `{{!mobileFirst}}` and then doing `{{mobileFirst}}` on mobile breakpoint?

Comment (dawnerd) [inline conf/default-breakpoints.yaml]: That'd work!

**Pull request merged**